unlock-app/src/__tests__/middlewares/lockMiddleware.test.js

```diff
@@ -334,15 +336,38 @@ describe('Lock middleware', () => {
     })
   })
 
-  it('it should handle error events triggered by the web3Service', () => {
-    const { store } = create()
-    mockWeb3Service.emit('error', { message: 'this was broken' })
-    expect(store.dispatch).toHaveBeenCalledWith(
-      expect.objectContaining({
-        type: SET_ERROR,
-        error: expect.anything(), // not sure how to test against jsx
+  describe('error events triggered by the web3Service', () => {
+    it('should handle error triggered when creating a lock', () => {
+      expect.assertions(2)
+      const { store } = create()
+      const transaction = {
+        type: TRANSACTION_TYPES.LOCK_CREATION,
+        lock: '0x123',
+      }
+      mockWeb3Service.emit('error', { message: 'this was broken' }, transaction)
+      expect(store.dispatch).toHaveBeenNthCalledWith(1, {
+        type: DELETE_LOCK,
+        address: transaction.lock,
       })
-    )
+      expect(store.dispatch).toHaveBeenNthCalledWith(
+        2,
+        expect.objectContaining({
+          type: SET_ERROR,
+          error: expect.anything(), // not sure how to test against jsx
+        })
+      )
+    })
+
+    it('it should handle error events triggered by the web3Service', () => {
+      const { store } = create()
+      mockWeb3Service.emit('error', { message: 'this was broken' })
+      expect(store.dispatch).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: SET_ERROR,
+          error: expect.anything(), // not sure how to test against jsx
```

Comment:
This would be solved with comments below about how to store and to note error conditions

Reply:
Good point!